addon/utils/array-helpers.js

```
@@ -193,3 +193,30 @@ export function findBy(arr, key, searchValue) {
 export function findById(arr, id) {
   return findBy(arr, 'id', id);
 }
+
+export function filterBy(arr, key, searchValue) {
+  if (arr !== null && typeof arr === 'object') {
+    if ('slice' in arr) {
+      arr = arr.slice();
+    }
+  }
+
+  if (!Array.isArray(arr)) {
```

should raise an Error or Exception instead.

see comment about `findBy()`.